File: src/Offering/Offering/Models/Localization.cs
namespace Offering.Models;

public class Localization
{
    public string CountryCode { get; private set; }
    
    public string CountryName { get; private set; }
    
    public string LocalName { get; private set; }
    
    public string Currency { get; private set; }
    
    protected Localization(string countryCode, string countryName, string localName, string currency)
    {
        CountryCode = countryCode;
        CountryName = countryName;
        LocalName = localName;
        Currency = currency;
    }
    
    public static Localization Create(string countryCode, string countryName, string localName, string currency)
    {
        return new Localization(countryCode, countryName, localName, currency);
    }
}

File: src/OfferSB/src/main/java/com/example/offersb/models/Localization.java
package com.example.offersb.models;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;


@Entity
@AllArgsConstructor
@NoArgsConstructor
public class Localization {

    @Id
    @Getter
    private String countryCode;

    @Getter
    private String countryName;

    @Getter
    private String localName;

    @Getter
    private String currency;

    public static Localization create(String countryCode, String countryName, String localName, String currency) {
        return new Localization(countryCode, countryName, localName, currency);
    }
}
